packages/dev/core/src/Meshes/mesh.ts

```diff
@@ -3093,107 +3031,126 @@ export class Mesh extends AbstractMesh implements IGetSetVerticesData {
-        const newdata: { [key: string]: Array<number> } = {};
-        let kindIndex: number;
-        let kind: string;
-        for (kindIndex = 0; kindIndex < kinds.length; kindIndex++) {
-            kind = kinds[kindIndex];
-            const vertexBuffer = <VertexBuffer>this.getVertexBuffer(kind);
-            vbs[kind] = vertexBuffer;
-            data[kind] = <FloatArray>vbs[kind].getData();
-            newdata[kind] = [];
+            if (flattenNormals && kind === VertexBuffer.NormalKind) {
+                const normals = this._getFlattenedNormals(indices, data[VertexBuffer.PositionKind]);
+                this.setVerticesData(VertexBuffer.NormalKind, normals, vertexBuffer.isUpdatable(), stride);
+            } else {
+                this.setVerticesData(kind, getDuplicatedVertices(data[kind], stride), vertexBuffer.isUpdatable(), stride);
+            }
         }
 
-        // Save previous submeshes
-        const previousSubmeshes = this.subMeshes.slice(0);
+        //Update morph targets
+        if (this.morphTargetManager) {
+            for (let targetIndex = 0; targetIndex < this.morphTargetManager.numTargets; targetIndex++) {
+                const target = this.morphTargetManager.getTarget(targetIndex);
 
-        const indices = <IndicesArray>this.getIndices();
-        const totalIndices = this.getTotalIndices();
+                const positions = target.getPositions()!;
+                target.setPositions(getDuplicatedVertices(positions, 3));
 
-        // Generating unique vertices per face
-        let index: number;
-        for (index = 0; index < totalIndices; index++) {
-            const vertexIndex = indices[index];
+                const normals = target.getNormals();
+                if (normals) {
+                   target.setNormals(flattenNormals ? this._getFlattenedNormals(indices, positions) : getDuplicatedVertices(normals, 3));
+                }
 
-            for (kindIndex = 0; kindIndex < kinds.length; kindIndex++) {
-                kind = kinds[kindIndex];
-                const stride = vbs[kind].getStrideSize();
+                const tangents = target.getTangents();
+                if (tangents) {
+                    target.setTangents(getDuplicatedVertices(tangents, 3));
+                }
 
-                for (let offset = 0; offset < stride; offset++) {
-                    newdata[kind].push(data[kind][vertexIndex * stride + offset]);
+                const uvs = target.getUVs();
+                if (uvs) {
+                    target.setUVs(getDuplicatedVertices(uvs, 2));
                 }
             }
+            this.morphTargetManager.synchronize();
         }
 
-        // Updating indices
-        for (index = 0; index < totalIndices; index += 3) {
-            indices[index] = index;
-            indices[index + 1] = index + 1;
-            indices[index + 2] = index + 2;
+        // Update indices 
+        if(updateIndices) {

```

Comment:
I noticed that both of the original methods updated the index buffer. I'm not sure why, although this forum post gives some clue.
https://forum.babylonjs.com/t/mesh-converttounindexedmesh-fails-to-remove-indices/13269/2?u=alexchuber

Should the index buffer be updated every time? Or should the indices not be updated at all?
(I have it under a flag for now.)

Reply:
@Popov72 @sebavan @deltakosh Thoughts?

Reply:
Also, we shouldn't merge it as is. If we are going to have a flag, it should be exposed to the public function.

Reply:
I think we should keep backward compatibiliy, so keep the index array with 0,1,2,3,... values.